src/vsr/replica.zig

```diff
@@ -783,65 +837,69 @@ pub fn ReplicaType(
             tracer.flush();
         }
 
-        // Pings are used:
-        //  - By clients, to learn about the current view.
-        //  - By replicas, to synchronise cluster time and to probe for network connectivity.
-        //
-        // In the second case we avoid setting the view to make sure pings can still be sent
-        // during view changes.
+        /// Pings are used by replicas to synchronise cluster time and to probe for network connectivity.
         fn on_ping(self: *Self, message: *const Message) void {
+            assert(message.header.command == .ping);
             if (self.status != .normal and self.status != .view_change) return;
 
-            assert(self.status == .normal or self.status == .view_change);
```

Comment:
One benefit of this assert is that it states the positive version of the invariant. That's why, when I write early return if's, I usually use `if (!(invariant))` form, even if pulling `!` inside can make the expression simpler. It's easy to read it as "`invariant` holds" then.